File: ios/HybridViewModelArtboardProperty.swift
import RiveRuntime

class HybridViewModelArtboardProperty: HybridViewModelArtboardPropertySpec {
  private let property: ArtboardPropertyType

  init(property: ArtboardPropertyType) {
    self.property = property
    super.init()
  }

  func set(artboard: (any HybridBindableArtboardSpec)?) throws {
    if let hybridArtboard = artboard as? HybridBindableArtboard {
      property.setValue(hybridArtboard.bindableArtboard)
    } else {
      property.setValue(nil)
    }
  }
}

File: android/src/legacy/java/com/margelo/nitro/rive/HybridViewModelArtboardProperty.kt
package com.margelo.nitro.rive

import androidx.annotation.Keep
import app.rive.runtime.kotlin.core.ViewModelArtboardProperty
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridViewModelArtboardProperty(private val property: ViewModelArtboardProperty) :
  HybridViewModelArtboardPropertySpec() {

  override val memorySize: Long
    get() = VIEW_MODEL_HYBRID_MEMORY_SIZE

  override fun set(artboard: HybridBindableArtboardSpec?) {
    val bindable = (artboard as? HybridBindableArtboard)?.bindableArtboard
    property.set(bindable)
  }
}
